doc/collision_detection.md

```diff
@@ -0,0 +1,212 @@
+Forge2D by adding [flame_forge2d](https://github.com/flame-engine/flame_forge2d) as a dependency.
+But if you have a simpler usecase and just want to check for collisions of components and improve
+the accuraccy of gestures, Flame's built-in collision detection will serve you very well.
+
+If you have the following needs you should at least consider to use
+[Forge2D](https://github.com/flame-engine/forge2d):
+ - Interacting realistic forces
+ - Particle systems that can interact with other bodies
+ - Joints between bodies
+ - Very many bodies at the same time
+
+It is a good idea to just use the Flame collision detection system if you on the other hand only
+need some of the following things (since it is slightly simpler to not involve Forge2D):
+ - The ability to act on when some of your components collide
+ - The ability to act on when your components collide with the screen boundaries
+ - Complex shapes to act as a hitbox for your component so that gestures will be more accurate
+ - Hitboxes that can tell what part of a component that collided with something
+
+The collision detection system supports three different types of shapes that you can build hitboxes
+from, these shapes are Polygon, Rectangle and Circle. A hitbox can be represented by many shapes to
+form the area which can be used to either detect collisions or whether it contains a point or not,
+the latter is very useful for accurate gesture detection. The collision detection does not handle
+what should happen when two hitboxes collide, so it is up to the user to implement what will happen
+when for example two position components have intersecting hitboxes.
+
+## Mixins
+### Hitbox
+The `Hitbox` mixin is mainly used for two things; to make detection of collisions with other
+hitboxes and gestures on top of your `PositionComponent`s more accurate. Say that you have a fairly
+round rock as a `SpriteComponent` for example, then you don't want to register input that is in the
+corner of the image where the rock is not displayed, since an image is always rectangular. Then you
+can use the `Hitbox` mixin to define a more accurate polygon (or another shape) for which the input
+should be within for the event to be registered on your component.
+
+You can add new shapes to the `Hitbox` just like they are added in the below `Collidable` example.
+
+### Collidable
+The `Collidable` mixin is added to a `PositionComponent` that has a `HitBox` and it is used for
+detecting collisions with other `Collidable`s. To make your component collidable you would start off
+like this:
+
+```dart
+class MyCollidable extends PositionComponent with Hitbox, Collidable {
+  MyCollidable (
+    // This could also be done in onLoad instead of in the constructor
+    final shape = HitboxPolygon([
+      Vector2(0, 1),
+      Vector2(1, 0),
+      Vector2(0, -1),
+      Vector2(-1, 0),
+    ]);
+    addShape(shape);
+  }
+}
+```
+
```

Comment:
```
**Note**: if you do not add a shape to a `Hitbox` component it will have no shape by default and it will never collide with anything. If you want the component to have a default rectangular shape that it had before it was `Hitbox` do xxxx
```

Reply:
Added on line 42